myth/Themers/LocalizedViewThemer.php

```diff
@@ -0,0 +1,1165 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @package     Sprint
+ * @author      Lonnie Ezell
+ * @copyright   Copyright 2014-2015, New Myth Media, LLC (http://newmythmedia.com)
+ * @license     http://opensource.org/licenses/MIT  (MIT)
+ * @link        http://sprintphp.com
+ * @since       Version 1.0
+ */
+
+use Myth\Modules as Modules;
+use Myth\Themers\ThemerInterface;
+
+class ViewThemer implements ThemerInterface
+{
+
+    protected $theme = '';
+
+    protected $default_theme = null;
+
+	protected $active_theme = null;
+
+    protected $layout = 'index';
+
+    protected $view = '';
+
+    protected $vars = [];
+
+    protected $folders = [];
+
+    protected $variants = [];
+
+    protected $current_variant = '';
+
+	protected $parse_views = false;
+
+    protected $ci;
+
+    //--------------------------------------------------------------------
+
+    public function __construct($ci)
+    {
+        $this->ci = $ci;
+
+	    $this->parse_views = config_item('theme.parse_views');
+    }
+
+    //--------------------------------------------------------------------
+
+    /**
+     * The main entryway into rendering a view. This is called from the
+     * controller and is generally the last method called.
+     *
+     * @param string $layout        If provided, will override the default layout.
+     * @param int    $cache_time    The number of seconds to cache the output for.
+     * @return mixed
+     */
+    public function render($layout = null)
+    {
+        // Make the template engine available within the views.
+        $this->vars['themer'] = $this;
+
+        // Render our current view content
+        $this->vars['view_content'] = $this->content();
+
+        $theme = empty($this->theme) ? $this->default_theme : $this->theme;
+
+        if (! isset($this->folders[$theme])) {
+            throw new \LogicException( sprintf( lang('theme.bad_folder'), $theme ) );
+        }
+
+	    $this->active_theme = $theme;
+
+        // Make the path available within views.
+        $this->vars['theme_path'] = $this->folders[$theme];
+
+        return $this->display($this->folders[$theme] . '/' . $this->layout);
```

Comment:
Is in this line necessary to add `$this->folders[$theme]`? As loadView will take care of this too:
```
private function loadView($view, $data)
{
// First - does it exist in the current theme?
$theme = ! empty($this->active_theme) ? $this->active_theme : $this->default_theme;
$theme = ! empty($this->folders[$theme]) ? $this->folders[$theme] : $theme;
```